File: partner-edit/frontend/src/app/api/model/dual-search-response.ts
import { PartnerGroupSearchDto } from './partner-group-search-dto';

export interface SearchEngineResult {
  results: Array<PartnerGroupSearchDto>;
  totalCount: number;
  returnedCount: number;
  durationMs: number;
}

export interface QuerySummary {
  tokens: string[];
}

export interface DualSearchResponse {
  postgres: SearchEngineResult;
  elasticsearch: SearchEngineResult;
  query: QuerySummary;
}

export interface QuadSearchResponse {
  spring: { postgres: SearchEngineResult | null; elasticsearch: SearchEngineResult | null };
  quarkus: { postgres: SearchEngineResult | null; elasticsearch: SearchEngineResult | null };
  quarkusAvailable: boolean;
}

File: partner-search/spring/src/main/java/com/example/partner/dto/DualSearchResponse.java
package com.example.partner.dto;

import com.example.partner.model.PartnerGroupSearchDto;

import java.util.List;

public record DualSearchResponse(
    SearchEngineResult postgres,
    SearchEngineResult elasticsearch,
    QuerySummary query
) {

    public record SearchEngineResult(
        List<PartnerGroupSearchDto> results,
        int totalCount,
        int returnedCount,
        long durationMs,
        int storeTotalCount
    ) {}

    public record QuerySummary(List<String> tokens) {}
}
